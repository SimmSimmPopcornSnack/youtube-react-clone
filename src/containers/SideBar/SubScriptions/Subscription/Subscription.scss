@import "../../../../styles/shared.scss";

.subscription {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;

    i.icon {
        color: $red;
    }

    .new-video-count {
        color: $grey;
    }
}
