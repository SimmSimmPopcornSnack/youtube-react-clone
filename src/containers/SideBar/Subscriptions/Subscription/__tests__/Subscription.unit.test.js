import React from "react";
import { shallow } from "enzyme";
import { Subscription } from "../Subscription";

describe("Subscription", () => {
    test("renders empty Subscription", () => {
        const wrapper = shallow(
            <Subscription/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    test("renders broadcasting Subscription", () => {
        const wrapper = shallow(
            <Subscription broadcasting label="Producitoncoder"/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    test("renders non-broadcasting Subscription with new videos", () => {
        const wrapper = shallow(
            <Subscription amountNewVideos={4} label="Productioncoder"/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
