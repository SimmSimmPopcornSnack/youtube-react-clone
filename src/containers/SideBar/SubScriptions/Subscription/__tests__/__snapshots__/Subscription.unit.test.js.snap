// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription renders broadcasting Subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Producitoncoder
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders empty Subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription renders non-broadcasting Subscription with new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
    <span
      className="new-video-count"
    >
      4
    </span>
  </div>
</MenuItem>
`;
